core/src/main/java/amino/run/oms/migrationdecision/MetricWatcher.java

```diff
@@ -0,0 +1,282 @@
+            logger.fine(String.format("Replica status : [%s]", replicaStat));
+            if (bestCandidates.size() <= MAX_TOP_CANDIDATE) {
+                bestCandidates.add(replicaStat.getKey());
+            }
+        }
+
+        // clear replica heap
+        replicaHeap.clear();
+        return bestCandidates;
+    }
+
+    private boolean triggerMigration(ReplicaID replicaID, InetSocketAddress kernelServer) {
+        MicroServiceID microServiceID = replicaID.getOID();
+
+        // start migration
+        try {
+            KernelOID koid = microServiceManager.getRootGroupId(microServiceID);
+            EventHandler groupPolicyEventHandler =
+                    microServiceManager.getGroupDispatcher(microServiceID, koid);
+            DefaultPolicy.DefaultGroupPolicy replicaGroupPolicy =
+                    (DefaultPolicy.DefaultGroupPolicy) groupPolicyEventHandler.getObjects().get(0);
+            replicaGroupPolicy.onNotification(new MigrationNotification(replicaID, kernelServer));
+        } catch (MicroServiceNotFoundException e) {
+            logger.warning(
+                    String.format(
+                            "Failed to migrate replica [%s] as micro service not available",
+                            replicaID));
+            e.printStackTrace();
+            return false;
+        } catch (RemoteException e) {
+            logger.warning(
+                    String.format(
+                            "Failed to migrate replica [%s] with remote exception : %s",
+                            replicaID, e.getMessage()));
+            e.printStackTrace();
+            return false;
+        }
+
+        return true;
+    }
+
+    private long aggregatedProcessingTime(ReplicaID replicaID)
+            throws MicroServiceNotFoundException, MicroServiceReplicaNotFoundException,
+                    KernelServerNotFoundException {
+        /*
+           To measure time consumption by vertex Vk we use the following formula
+           tk = Tk,posk+i + summation { (Ski+Sik)/Rposk,posi + Lposk,posi }
+        */
+
+        // get current kernel server for replica
+        InetSocketAddress replicaKernelServer =
+                microServiceManager.getReplicaDispatcher(replicaID).getHost();
+        // get metric stat for current replica kernel server
+        Map<InetSocketAddress, NodeMetric> replicaNodeMetric =
+                serverManager.getKernelServerMetric(replicaKernelServer);
+
+        // get metrics stat of RPC processing time from different kernel client
+        // TODO currently considering processing time from different kernel client for simplicity.
+        //  Need to get processing time within micro service replica
+        Map<InetSocketAddress, RPCMetric> replicaMetrics =
+                microServiceManager.getMicroServiceMetric(replicaID);
+
+        // calculate time factor impact on RPC with latency and bandwidth
+        long averageExecutionTime = 0;
```

Comment:
Needs to be renamed to avgElapsedTime to be consistent with replicaMetric.getValue().elapsedTime